presto-common/src/main/java/com/facebook/presto/common/block/ByteArrayBlockEncoding.java

```diff
@@ -49,16 +51,27 @@ public void writeBlock(BlockEncodingSerde blockEncodingSerde, SliceOutput sliceO
     public Block readBlock(BlockEncodingSerde blockEncodingSerde, SliceInput sliceInput)
     {
         int positionCount = sliceInput.readInt();
-
-        boolean[] valueIsNull = decodeNullBits(sliceInput, positionCount).orElse(null);
-
         byte[] values = new byte[positionCount];
-        for (int position = 0; position < positionCount; position++) {
-            if (valueIsNull == null || !valueIsNull[position]) {
-                values[position] = sliceInput.readByte();
-            }
+
+        // Fast track if no nulls present
+        if (!sliceInput.readBoolean()) {
+            sliceInput.readBytes(wrappedBuffer(values));
```

Comment:
There is no need to create a new Slice object by wrappedBuffer. There is `public final void readBytes(byte[] destination)` in SliceInput.